doc/mgr/dashboard.rst

```diff
@@ -298,6 +301,171 @@ previously defined username and password. Select the **Keep me logged in**
+
+A *role* specifies a set of mappings between a *security scope* and a set of
+*permissions*. There are four types of permissions:
+
+- **read**
+- **create**
+- **update**
+- **delete**
+
+Below we show an example of a role specification based on a Python dictionary::
+
+  # example of a role
+  {
+    'role': 'my_new_role',
+    'scopes_permissions': {
+      'pool': ['read', 'create'],
+      'rbd-image': ['read', 'create', 'update', 'delete']
+    }
+  }
+
+The above role dictates that a user has *read* and *create* permissions for
+features related with pool management, and has full permissions for
+features related with RBD image management.
+
+The Dashboard already provides a set of predefined roles that we call
+*system roles*, and can be used right away in a fresh Ceph Dashboard
+installation.
+
+The list of system roles are:
+
+- **Administrator**: provides full permissions for all security scopes.
+- **Read-only**: provides *read* permission for all security scopes.
+- **Block Manager**: provides full permissions for *rbd-image*,
+  *rbd-mirroring*, and *iscsi* scopes.
+- **RGW Manager**: provides full permissions for the *rgw* scope
+- **Cluster Manager**: provides full permissions for the *hosts*, *osd*,
+  *monitor*, *manager*, and *config-opt* scopes.
+- **Pool Manager**: provides full permissions for the *pool* scope.
+- **CephFS Manager**: provides full permissions for the *cephfs* scope.
+
+The list of currently available roles can be retrived by the following
+command::
+
+  $ ceph dashboard ac-role-show [<rolename>]
+
+It is also possible to create new roles using CLI commands. The available
+commands to manage roles are the following:
+
+- *Create Role*::
+
+  $ ceph dashboard ac-role-create <rolename>
+
+- *Delete Role*::
+
+  $ ceph dashboard ac-role-delete <rolename>
+
+- *Add Scope Permissions to Role*::
+
+  $ ceph dashboard ac-role-add-scope-perms <rolename> <scopename> <permission> [<permission>...]
+
+- *Delete Scope Permission from Role*::
+
+  $ ceph dashboard ac-role-del-perms <rolename> <scopename>
+
+To associate roles to users, the following CLI commands are available:
+
+- *Set User Roles*::
+
+  $ ceph dashboard ac-user-set-roles <username> <rolename> [<rolename>...]
+
+- *Add Roles To User*::
+
+  $ ceph dashboard ac-user-add-roles <username> <rolename> [<rolename>...]
+
+- *Delete Roles from User*::
+
+  $ ceph dashboard ac-user-del-roles <username> <rolename> [<rolename>...]
+
+
+Example of user and custom role creation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this section we show a full example of the commands that need to be used
+in order to create a user account, that should be able to manage RBD images,
+view and create Ceph pools, and have read-only access to any other scopes.
+
+1. *Create the user*::
+
+   $ ceph dashboard ac-user-create bob mypassword
+
+2. *Create role and specify scope permissions*::
+
+   $ ceph dashboard ac-role-create "rbd/pool manager"
```

Comment:
`ceph dashboard ac-role-create "rbd/pool manager"` fails with
```
Invalid command: unused arguments: [u'manager']
dashboard ac-role-create <rolename> :  Create a new access control role
Error EINVAL: invalid command
```
although I used quotes.

Reply:
Hmm, `ceph dashboard ac-role-create "rbd/pool manager"` worked for me:
```
# ./bin/ceph dashboard ac-role-create "rbd/pool manager"
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
2018-06-14 08:49:00.393 7f3f72ec9700 -1 WARNING: all dangerous and experimental features are enabled.
2018-06-14 08:49:00.408 7f3f72ec9700 -1 WARNING: all dangerous and experimental features are enabled.
{"scopes_permissions": {}, "name": "rbd/pool manager"}
```